src/plots/plots.js

```diff
@@ -1076,10 +1079,10 @@ plots.supplyFrameDefaults = function(frameIn) {
     return frameOut;
 };
 
-plots.supplyTraceDefaults = function(traceIn, colorIndex, layout, traceInIndex) {
+plots.supplyTraceDefaults = function(traceIn, colorIndex, layout, traceInIndex, uid) {
     var colorway = layout.colorway || Color.defaults;
-    var traceOut = {},
-        defaultColor = colorway[colorIndex % colorway.length];
+    var traceOut = {uid: uid};
```

Comment:
Or alternatively, I think we could move this line:

![image](https://user-images.githubusercontent.com/6675409/40743247-fa945b26-641e-11e8-8a2e-92b5fe585f53.png)

to just before this line:

![image](https://user-images.githubusercontent.com/6675409/40743266-07c4e130-641f-11e8-987c-e7c28ef14adc.png)

Reply:
the above comment is non-blocking.

Reply:
yeah probably - but I thought it would be better in `supplyDefaults` so we don't add another order-dependence to the `plot` step - carpet dependents need this too. I think they're already required to `plot` after the carpet for other reasons but still...

Reply:
> but I thought it would be better in supplyDefaults so we don't add another order-dependence to the plot step - carpet dependents need this too.

Good point. Your way is :ok_hand: